src/plugins/data/public/search/search_source/search_source.ts

```diff
@@ -465,7 +465,7 @@ export class SearchSource {
 
       // request the remaining fields from both stored_fields and _source
       const remainingFields = difference(fields, keys(body.script_fields));
-      body.stored_fields = remainingFields;
+      body.fields = remainingFields;
       setWith(body, '_source.includes', remainingFields, (nsValue) =>
```

Comment:
dear @jtibshirani , with this change Discover via the underlying search service would use the fields API, and would therefore support runtime fields. Question is:  could there be fields available in `stored_fields` that aren't in the `fields` API? thx!

Reply:
@kertal `stored_fields` are not included in the 'fields' option. So yes, there could be fields available in `stored_fields` that do not come back in `fields`.

Reply:
@jtibshirani thx, i've solved it differently now!